pdns/ueberbackend.cc

```diff
@@ -502,48 +502,81 @@ int UeberBackend::cacheHas(const Question &q, vector<DNSZoneRecord> &rrs)
 {
   extern AuthQueryCache QC;
 
-  if(!d_cache_ttl && ! d_negcache_ttl) {
+  if (!d_cache_ttl && !d_negcache_ttl) {
     return -1;
   }
 
   rrs.clear();
-  //  g_log<<Logger::Warning<<"looking up: '"<<q.qname+"'|N|"+q.qtype.getName()+"|"+itoa(q.zoneId)<<endl;
+  //  g_log<<Logger::Warning<<"looking up: '"<<q.qname<<"'|N|"<<q.qtype.getName()<<"|"<<q.zoneId<<endl;
 
-  bool ret=QC.getEntry(q.qname, q.qtype, rrs, q.zoneId);   // think about lowercasing here
-  if(!ret) {
-    return -1;
+  std::vector<DNSZoneRecord> anyRecs;
+  bool ret = QC.getEntry(q.qname, QType::ANY, anyRecs, q.zoneId);
+
+  if (ret) {
+    if (anyRecs.empty()) {// negatively cached
+      return 0;
+    }
+
+    for (auto& rec : anyRecs) {
+      if (q.qtype.getCode() == QType::ANY || rec.dr.d_type == q.qtype.getCode()) {
+        rrs.push_back(std::move(rec));
```

Comment:
emplace_back?

Reply:
Won't the move-constructor be called in the exact same way in both cases?

Reply:
possibly. i just wonder if it could be rrs.emplace_back(rec)?

Reply:
That would allocate and copy two `DNSName`s and one `shared_ptr<DNSRecordContent>` instead of moving them, so I don't think that would be better.